include/libtorrent/aux_/export.hpp

```diff
@@ -100,7 +100,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 #if !defined TORRENT_EXPORT_EXTRA \
-  && ((defined __GNUC__ && __GNUC__ >= 4) || defined __clang__)
+  && ((defined __GNUC__ && __GNUC__ >= 4) || defined __clang__) && !defined _WIN32
```

Comment:
I would appreciate a comment explaining that the problematic case is clang on windows. I suppose `_WIN32` is a good proxy for this, although `_WIN32` indicates a platform, not a compiler. but the interaction with shared libraries is platform specific anyway